webhooks/cluster_test.go

```diff
@@ -278,9 +283,13 @@ func TestClusterTopologyValidation(t *testing.T) {
 					WithClass("foo").
 					WithVersion("v1.19.1").
 					WithMachineDeployment(
-						builder.MachineDeploymentTopology("aa").Build()).
+						builder.MachineDeploymentTopology("workers1").
+							WithClass("aa").
+							Build()).
 					WithMachineDeployment(
-						builder.MachineDeploymentTopology("bb").Build()).
+						builder.MachineDeploymentTopology("workers2").
+							WithClass("bb").
+							Build()).
 					Build()).
 				Build(),
 		},
```

Comment:
Can you check the test case in l.297: `should return error on update when Topology class is changed`

I don't think that should return an error

Reply:
I think it currently returns the error because the ClusterClass don't exist

Reply:
Note: Maybe we should merge this table test with `TestClusterTopologyValidationWithClient` now (or move some of the test cases from this test down to `TestClusterTopologyValidationWithClient`)

Reply:
You're right about this test - and it already overlaps with the non-existent class test in TestClusterTopologyValidationWithClient so I'll remove it.